include/xgboost/gbm.h

```diff
@@ -10,6 +10,7 @@
 
 #include <dmlc/registry.h>
 #include <xgboost/base.h>
+#include <xgboost/json.h>
```

Comment:
Do we really need to include json.h here? It seems like a forward declaration of `class Json;` should be enough

Reply:
A friendly ping. A forward definition help to reduce code interdependence and decrease a compile time.
It is not a major issue but if a forward declaration is indeed enough I would rather use it

Reply:
Done.

Reply:
I do not want to nitpick. I am just a bit confused. It seems this include is still here. Do I miss some other pull request or updates?

Reply:
It's removed right now, along with other headers.